portfolio/src/main/java/com/google/sps/servlets/BlobstoreServlet.java

```diff
@@ -0,0 +1,93 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+
+import com.google.gson.Gson;
+import java.util.Map;
+import java.util.List;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+
+/** Servlet to handle blobstore */
+@WebServlet("/blobstore")
+public class BlobstoreServlet extends HttpServlet {
+
+    BlobstoreService blobstore;
+
+    @Override
+    public void init() {
+        this.blobstore = BlobstoreServiceFactory.getBlobstoreService();
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String uploadUrl = this.blobstore.createUploadUrl("/blobstore");
+
+        response.setContentType("application/json;");
+        Gson gson = new Gson();
+        response.getWriter().println(gson.toJson(uploadUrl));
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String action = request.getParameter("action");
+        String imageUrl = getUploadedFileUrl(request, "image");
+
+        RequestDispatcher rd = request.getRequestDispatcher(action);
+        request.setAttribute("imageUrl", imageUrl);
+        try {
+            rd.forward(request, response);
+        }
+        catch(ServletException se) {
+            se.printStackTrace();
```

Comment:
Do not swallow exception, unless it is a valid reason. This is considered bad practice and would normally introduce bugs.
If you don't want to handle it, you can mark the method to `throws`.

Besides, can explain why you want to to a forward?
Actually I think you may want to submit the POST request as part of the /comments.